rope/base/utils.py

```diff
@@ -81,3 +81,48 @@ def __call__(self, *args, **kwds):
         if len(self.cache) > self.count:
             del self.cache[0]
         return result
+
+class memoize(object):
```

Comment:
[There is](https://github.com/python-rope/rope/compare/master...emacsway:cache_optimization) how I see it currently. But there is also [minor FIXME](https://github.com/emacsway/rope/blob/70dc6bbb296489a81b2735d6c007758464a99af1/rope/base/utils.py#L87).
The easiest way is the using FIFO instead of LRU, i.e. just remove [these lines](https://github.com/emacsway/rope/blob/70dc6bbb296489a81b2735d6c007758464a99af1/rope/base/utils.py#L86-L88). I have not idea currently how we can remove iteration and save LRU.